src/pytest_memray/plugin.py

```diff
@@ -138,8 +178,10 @@ def _build_bin_path() -> Path:
         def wrapper(*args: Any, **kwargs: Any) -> object | None:
             try:
                 result_file = _build_bin_path()
-                with Tracker(result_file):
-                    result: object | None = func(*args, **kwargs)
+                runner_function = functools.partial(func, *args, **kwargs)
+                result: object | None = runner(
```

Comment:
This was in the old code, but - isn't `object | None` effectively the same as just `object`, since `None` is a kind of object?